api/data_refinery_api/tests.py

```diff
@@ -631,9 +631,10 @@ def test_dataset_adding_non_downloadable_samples_fails(self):
             content_type="application/json",
         )
         self.assertEqual(response.status_code, 400)
-        self.assertEqual(
-            response.json()["non_field_errors"][0], "Non-downloadable sample(s) '456' in dataset"
+        self.assertIn(
+            "Non-downloadable sample(s) in dataset", response.json()["message"][0],
```

Comment:
Why is the `[0]` necessary? Is it returning a list of strings instead of just a single string?

Reply:
Yeah, for some reason Django Rest Framework is transforming the string into a list of strings of length 1